lib/Target/ExportSMTLIB/ExportSMTLIB.cpp

```diff
@@ -334,9 +337,21 @@ struct ExpressionVisitor
     newInfo.stream.indent(newInfo.indentLevel);
     if (failed(printExpression(worklist, newInfo)))
       return failure();
+
     info.stream << info.valueMap.lookup(yieldedValue);
-    for (unsigned k = 0; k < newInfo.openParens + 1; ++k)
-      info.stream << ")";
+
+    if(weight != 0 && newInfo.openParens > 0){
+      for(int j=0;j<newInfo.openParens;j++)
```

Comment:
Super-nit: the LLVM codebase typically prefers `++j` instead of `j++`

Reply:
oh cool! I did not know :)